src/utils/save-file.ts

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2022, Nuance, Inc. and its contributors.
+ * All rights reserved.
+ *
+ * This source code is licensed under the Apache-2.0 license found in
+ * the LICENSE file in the root directory of this source tree.
+ */
+
+/** eslint-disable unicorn/prefer-node-protocol */
+import makeDebug from 'debug'
+import {writeFile} from 'node:fs'
```

Comment:
import and use `writeFileSync` instead otherwise, you will have to deal with async code later below.